File: lib/src/main/java/com/cgbsoft/lib/utils/tools/NetUtils.java
package com.cgbsoft.lib.utils.tools;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.telephony.TelephonyManager;

import com.cgbsoft.lib.BaseApplication;

/**
 * 获取网络连接状态
 * Created by xiaoyu.zhang on 2016/12/6 13:22
 * Email:zhangxyfs@126.com
 *  
 */
public class NetUtils {
    /**
     * 网络是否可用
     * @param context
     * @return
     */
    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivity = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivity != null) {
            NetworkInfo info = connectivity.getActiveNetworkInfo();
            if (info != null && info.isConnected())
            {
                // 当前网络是连接的
                if (info.getState() == NetworkInfo.State.CONNECTED)
                {
                    // 当前所连接的网络可用
                    return true;
                }
            }

        }
        PromptManager.ShowCustomToast(context,"网络断开请连接网络");
        return false;
    }
    public static NetState getNetState() {
        NetState stateCode = NetState.NET_NO;
        ConnectivityManager cm = (ConnectivityManager) BaseApplication.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        if (ni != null && ni.isConnectedOrConnecting()) {
            switch (ni.getType()) {
                case ConnectivityManager.TYPE_WIFI:
                    stateCode = NetState.NET_WIFI;
                    break;
                case ConnectivityManager.TYPE_MOBILE:
                    switch (ni.getSubtype()) {
                        case TelephonyManager.NETWORK_TYPE_GPRS: //联通2g
                        case TelephonyManager.NETWORK_TYPE_CDMA: //电信2g
                        case TelephonyManager.NETWORK_TYPE_EDGE: //移动2g
                        case TelephonyManager.NETWORK_TYPE_1xRTT:
                        case TelephonyManager.NETWORK_TYPE_IDEN:
                            stateCode = NetState.NET_2G;
                            break;
                        case TelephonyManager.NETWORK_TYPE_EVDO_A: //电信3g
                        case TelephonyManager.NETWORK_TYPE_UMTS:
                        case TelephonyManager.NETWORK_TYPE_EVDO_0:
                        case TelephonyManager.NETWORK_TYPE_HSDPA:
                        case TelephonyManager.NETWORK_TYPE_HSUPA:
                        case TelephonyManager.NETWORK_TYPE_HSPA:
                        case TelephonyManager.NETWORK_TYPE_EVDO_B:
                        case TelephonyManager.NETWORK_TYPE_EHRPD:
                        case TelephonyManager.NETWORK_TYPE_HSPAP:
                            stateCode = NetState.NET_3G;
                            break;
                        case TelephonyManager.NETWORK_TYPE_LTE:
                            stateCode = NetState.NET_4G;
                            break;
                        default:
                            stateCode = NetState.NET_UNKNOWN;
                    }
                    break;
                default:
                    stateCode = NetState.NET_UNKNOWN;
            }

        }
        return stateCode;
    }

    public enum NetState {
        NET_NO,//无网络
        NET_2G,
        NET_3G,
        NET_4G,
        NET_WIFI,
        NET_UNKNOWN//未知
    }
}

File: enter/src/main/java/app/ndk/com/enter/test/NetUtils.kt
package app.ndk.com.enter.test

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.telephony.TelephonyManager

/**
 *desc  ${DESC}
 *author wangyongkui  wangyongkui@simuyun.com
 *日期 2017/8/21-16:42
 */
class NetUtils(val context: Context) {
    /**
     * 手机管理器
     */
    val TelephonyManger = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
    /**
     * 链接活动管理器
     */
    val ConnectManger = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

    /**
     * 网络状态
     */
    val Netenable: Boolean get() {
//        val connManger = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info = ConnectManger.activeNetworkInfo
        return NetworkInfo.State.CONNECTED == info.state
    }
    /**
     * wifi是否可用
     */
    val WifiAvailable: Boolean
        get() {
            val inf = ConnectManger.activeNetworkInfo
            return (null != inf && inf.isConnected && inf.type == ConnectivityManager.TYPE_WIFI)
        }




}